File: Java/FacadePattern/src/facade/Facade.java
package facade;

import cluster_of_classes.Audi;
import cluster_of_classes.Car;
import cluster_of_classes.Honda;
import cluster_of_classes.Toyota;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 04-06-2021
 * Time: 03:06 AM
 */
public class Facade {

    private final Car audi;
    private final Car honda;
    private final Car toyota;

    public Facade() {
        this.audi = new Audi();
        this.honda = new Honda();
        this.toyota = new Toyota();
    }

    public void getAudi() {
        audi.model();
        audi.price();
    }

    public void getHonda() {
        honda.model();
        honda.price();

    }

    public void getToyota() {
        toyota.model();
        toyota.price();
    }
}

File: Kotlin/FacadePattern/src/facade/Facade.kt
package facade

import cluster_of_classes.Audi
import cluster_of_classes.Honda
import cluster_of_classes.Toyota

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 04-06-2021
 *   Time: 03:29 AM
 *
 */
class Facade {
    private var audi = Audi()
    private var honda = Honda()
    private var toyota = Toyota()

    fun buyAudi(){
        audi.model()
        audi.price()
    }

    fun buyHonda(){
        honda.model()
        honda.price()
    }

    fun buyToyota(){
        toyota.model()
        toyota.price()
    }

}